03-viz.Rmd

```diff
@@ -1,40 +1,528 @@
+```
+
+```{r 03-data-morley-hist-3-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(morley_hist)
+```
+
+Unfortunately, the attempt to separate out the experiment number visually has created a bit of a mess. All of the colours
+are blending together, and although it is possible to derive *some* insight from this (e.g., experiments 1 and 3 had some 
+of the most incorrect measurements), it isn't the clearest way to convey our message and answer the question. Let's try a different strategy of 
+creating multiple separate histograms on top of one another. 
+
+In order to create a plot in `ggplot2` that has multiple subplots arranged in a grid, we use the `facet_grid` function.
+The argument to `facet_grid` specifies the variable(s) used to split the plot into subplots. It has the syntax `vertical_variable ~ horizontal_variable`,
+where `veritcal_variable` is used to split the plot vertically, `horizontal_variable` is used to split horizontally, and `.` is used if there should be no split
+along that axis. In our case
+we only want to split vertically along the `Expt` variable, so we use `Expt ~ .` as the argument to `facet_grid`.
+
+```{r 03-data-morley-hist-4}
+morley_hist <- ggplot(morley, aes(x = Speed, fill=factor(Expt))) + 
+                 geom_histogram(position="identity") +
+                 facet_grid(Expt ~ .) +
+                 geom_vline(xintercept = 792.458, linetype="dashed", size=1.0)
+```
+
+```{r 03-data-morley-hist-4-print, echo = FALSE, results = 'hide', message = FALSE, warning = FALSE}
+print(morley_hist)
+```
+
+The visualization now makes it quite clear how accurate the different experiments were with respect to one another. 
+There are two finishing touches to make this visualization even clearer. First and foremost, we need to add informative axis labels
+using the `labs` function, and increase the font size to make it readable using the `theme` function. Second, and perhaps more subtly, even though it 
+is easy to compare the experiments on this plot to one another, it is hard to get a sense 
+for just how accurate all the experiments were overall. For example, how accurate is the value 800 on the plot, relative to the true speed of light?
+To answer this question we'll use the `mutate` function to transform our data into a relative measure of accuracy rather than absolute measurements:
+```{r 03-data-morley-hist-5}
+morley_rel <- mutate(morley, relative_accuracy = 100*( (299000 + Speed) - 299792.458 ) / (299792.458))
+morley_hist <- ggplot(morley_rel, aes(x = relative_accuracy, fill=factor(Expt))) + 
+                 geom_histogram(position="identity") +
+                 facet_grid(Expt ~ .) +
+                 geom_vline(xintercept = 0, linetype="dashed", size=1.0) + 
+                 labs(x = 'Relative Accuracy (%)', y = '# Measurements', fill = 'Experiment ID') + 
+                 theme(text = element_text(size = 18))
```

Comment:
The y-axis values are too jumbled... Maybe have less of them, or make the plot taller?